agent/taskresource/envFiles/envfile.go

```diff
@@ -402,7 +407,10 @@ func (envfile *EnvironmentFileResource) writeEnvFile(writeFunc func(file oswrapp
 		return err
 	}
 
-	err = rename(tmpFile.Name(), fullPathName)
+	backoff := retry.NewExponentialBackoff(renameBackoffMin, renameBackoffMax, renameBackoffJitter, renameBackoffMultiple)
+	err = retry.RetryWithBackoff(backoff, func() error {
```

Comment:
can we update the unit test accordingly? I was reading the code of `retry.RetryWithBackoff` and if there is in fact an error with `rename`, the returning error won't be of type `apierrors.Retriable`. I think in case there's some unrecoverable error, this function will be stuck for ever.

Reply:
Yes, I just saw the unit tests failing, I will update in new revision.